### PR TITLE
Switch to Mambaforge in Github Actions tests

### PR DESCRIPTION
**Pull request by valeriupredoi**
## Description

~Hold your horses :horse: - I'll add info here and switch it to Ready for Review once I can activate the GA tests on this branch - currently am waiting for another batch of GA tests to finish, so we don't overload and slow down that poor machine~

This is done now: changes fully implemented and tested: 43 tests ran in total (holy!)
@zklaus thanks again for the changes in Tool, I merely reproduced/ported them here, note however, that I've not changed the conda lockfile generator and validator - that one uses conda-forge fully (and not any `defaults` guff) and I think we should keep it that way. Any opinions on that? Oh and the PyPi package builder should be unchanged too. Have a look at the tests too, if you OK with the changes I'll remove the feature branch from the GA test workflow scripts.

***

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by schlunma** [comment]
@valeriupredoi you reckon we get this in in time for v2.5? Today would be a good day to merge this :smile:

**Comment by valeriupredoi** [comment]
yes man! I am gonna solve the conflicts now, cheers for reminding me :beer:

**Comment by valeriupredoi** [comment]
ok done, also removed the curr feature branch from all the GA tests, they passed fine (see photographic evidence :laughing: ) 
testsBranch

File: .github/workflows/run-tests-monitor.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Monitor Tests
 on:
   push:
     branches:
-      - main
+    - main
```
Comment (zklaus) [inline .github/workflows/run-tests-monitor.yml:6]: Help, we lost proper indentation again
```suggestion
      - main
```

**Review by zklaus — CHANGES_REQUESTED** [review]
Just a mini change request. Thanks, @valeriupredoi !

File: .github/workflows/run-tests-monitor.yml (hunk 1 above)
Comment (valeriupredoi) [reply .github/workflows/run-tests-monitor.yml]: thanks, man! :beer:

**Review by zklaus — APPROVED** [review]

**Comment by valeriupredoi** [comment]
cheers Klaus, you're a :star:

**Comment by zklaus** [comment]
No you are! :beers:

**Comment by valeriupredoi** [comment]
> No you are! beers

shucks :blush:

**Pull request merged**